### PR TITLE
Optimize much

### PR DESCRIPTION
**Pull request by Goooler**
- Update dependencies & Replace jcenter.
- Add .gitattributes & .editorconfig.
- Code cleanups.
- Add CI for building.

### CODE REVIEW

**Comment by ikarus23** [comment]
Thank you! I will have a closer look and give some feedback in the respective commit. A lot of my feedback will probably be questions, because I'm no Android developer guru like you ;)

**Comment by Goooler** [comment]
No problem.

**Review by ikarus23 — COMMENTED** [review]
I have no experience with github CI. I guess it will work since you have tested it :)
There is a new line missing at the end of the file.

File: Mifare Classic Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java [hunk 1]
```diff
@@ -906,18 +903,7 @@ public static int checkMifareClassicSupport(Tag tag, Context context) {
...
-                        } else {
-                            // MIFARE Classic 1k
-                            // MIFARE SmartMX 1k
-                            // MIFARE Plus S 2K SL1
-                            // MIFARE Plus X 2K SL1
-                            // MIFARE Plus SE 1K
-                            // MIFARE Plus EV1 2K/4K SL1
-                            return -1;
-                        }
+                        return -1;
```
Comment (ikarus23) [inline Mifare Classic Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java:906]: I did this to make the code more readable and closer to the datasheet linked in an above comment.
I don't like the optimized version. It is not readable.

File: Mifare Classic Tool/app/build.gradle [hunk 2]
```diff
@@ -4,7 +4,6 @@ plugins {
 
 android {
     compileSdkVersion 31
-    buildToolsVersion '31.0.0'
```
Comment (ikarus23) [inline Mifare Classic Tool/app/build.gradle:7]: Is safe to remove this? What build tools version will be used when it's not declared?

File: Mifare Classic Tool/gradlew [hunk 3]
```diff
@@ -1,79 +1,129 @@
-#!/usr/bin/env bash
```
Comment (ikarus23) [inline Mifare Classic Tool/gradlew:1]: Why replace it with just `/bin/sh`? The old format made more sense to me.
Did you do the big grade update yourself or is there an automatic tool for it?

**Review by ikarus23 — COMMENTED** [review]
Looks good in most places, but some small changes are required.

File: Mifare Classic Tool/app/build.gradle (hunk 2 above)
Comment (Goooler) [reply Mifare Classic Tool/app/build.gradle:7]: Latest.

File: Mifare Classic Tool/gradlew (hunk 3 above)
Comment (Goooler) [reply Mifare Classic Tool/gradlew:1]: Generate gradle wrapper by running `./gradlew wrapper`.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle/wrapper/gradle-wrapper.properties', '**/buildSrc/src/main/kotlin/**.kt') }}
+      - name: Build
+        working-directory: ${{ env.PROJECT_PATH }}
+        run: ./gradlew --parallel assembleRelease
```
[the remark this answers is not in the document]
Comment (Goooler) [reply .github/workflows/ci.yml]: Added.

File: Mifare Classic Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java (hunk 1 above)
Comment (Goooler) [reply Mifare Classic Tool/app/src/main/java/de/syss/MifareClassicTool/Common.java]: Reverted.

**Comment by Goooler** [comment]
CI passed on my forked repo.

image

**Comment by ikarus23** [comment]
There was a big jar file at `Mifare Classic Tool/gradle/wrapper/gradle-wrapper.jar`. Should that be really part of the repo?

**Comment by ikarus23** [comment]
Well, I guess it should. At least it was there before. Just got bigger :)

**Comment by Goooler** [comment]
Yeah, you can learn the gradle project structure.

**Pull request merged**